lambda.tf

```
@@ -38,29 +38,29 @@ data "aws_iam_policy_document" "function" {
     effect = "Allow"
   }
 
-  # dynamic "statement" {
-  #   for_each = length(each.value.trigger.queue) > 0 ? { enabled = true } : {}
-  #   content {
-  #     sid       = "SQSTriggers"
-  #     resources = [for queue_key in keys(each.value.trigger.queue) : aws_sqs_queue.this[queue_key].arn]
-  #     effect    = "Allow"
-  #     actions = [
-  #       "sqs:ReceiveMessage",
-  #       "sqs:DeleteMessage",
-  #       "sqs:GetQueueAttributes"
-  #     ]
-  #   }
-  # }
-
-  # dynamic "statement" {
-  #   for_each = length(each.value.target.queue) > 0 ? { enabled = true } : {}
-  #   content {
-  #     sid       = "SQSTargets"
-  #     resources = [for queue_key in keys(each.value.target.queue) : aws_sqs_queue.this[queue_key].arn]
-  #     effect    = "Allow"
-  #     actions   = ["sqs:SendMessage"]
-  #   }
-  # }
+  dynamic "statement" {
+    for_each = length(each.value.trigger.queue) > 0 ? { enabled = true } : {}
+    content {
+      sid       = "SQSTriggers"
+      resources = [for queue_key in keys(each.value.trigger.queue) : aws_sqs_queue.this[queue_key].arn]
+      effect    = "Allow"
+      actions = [
+        "sqs:ReceiveMessage",
+        "sqs:DeleteMessage",
+        "sqs:GetQueueAttributes"
+      ]
+    }
+  }
+
+  dynamic "statement" {
+    for_each = length(each.value.target.queue) > 0 ? { enabled = true } : {}
+    content {
+      sid       = "SQSTargets"
+      resources = [for queue_key in keys(each.value.target.queue) : aws_sqs_queue.this[queue_key].arn]
+      effect    = "Allow"
+      actions   = ["sqs:SendMessage"]
+    }
+  }
```

```suggestion
  dynamic "statement" {
    for_each = { trigger = each.value.trigger.queue }
    content {
      sid       = "SQSTriggers"
      resources = [for queue_key in keys(each.value) : aws_sqs_queue.this[queue_key].arn]
      effect    = "Allow"
      actions = [
        "sqs:ReceiveMessage",
        "sqs:DeleteMessage",
        "sqs:GetQueueAttributes"
      ]
    }
  }

  dynamic "statement" {
    for_each = { target = each.value.target.queue }
    content {
      sid       = "SQSTargets"
      resources = [for queue_key in keys(each.value) : aws_sqs_queue.this[queue_key].arn]
      effect    = "Allow"
      actions   = ["sqs:SendMessage"]
    }
  }
```
Something along those lines should work, and be a little more readable IMO

---

nvm